cluster/images/etcd-version-monitor/etcd-version-monitor.go

```diff
@@ -58,15 +59,16 @@ const (
 // Initialize prometheus metrics to be exported.
 var (
 	// Register all custom metrics with a dedicated registry to keep them separate.
-	customMetricRegistry = prometheus.NewRegistry()
+	customMetricRegistry = metrics.NewKubeRegistry()
 
 	// Custom etcd version metric since etcd 3.2- does not export one.
 	// This will be replaced by https://github.com/coreos/etcd/pull/8960 in etcd 3.3.
-	etcdVersion = prometheus.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Name:      "version_info",
-			Help:      "Etcd server's binary version",
+	etcdVersion = metrics.NewGaugeVec(
```

Comment:
This would be better as a GaugeFunc no? This way if the etcd cluster is upgraded we will end up with multiple active time-series reporting multiple versions, when there is really only one.

Reply:
There is actually already logic for this in place (https://github.com/kubernetes/kubernetes/blob/7cbd6bf9eae6512f0bffa85c438100640692fe9b/cluster/images/etcd-version-monitor/etcd-version-monitor.go#L232). 

Also, not sure that a gaugeFunc would help in that respect (doesn't it just allow for lazily evaluating the metric value during scrape?).

Reply:
Makes sense. Sounds good to me this way as well then.